### ISSUE TITLE
Assemblies not being included when building Godot4.3 C# build

### ISSUE THREAD

**Issue by blockimperiumdao** [godotengine/godot#94436]
### Tested versions

- Reproducible in v4.3.beta3.mono.official [82cedc83c]

### System information

Godot v4.3.beta3.mono - macOS 15.0.0 - Vulkan (Forward+) - integrated Apple M2 Pro - Apple M2 Pro (10 Threads)

### Issue description
```
================================================================
Godot Engine v4.3.beta3.mono.official.82cedc83c - https://godotengine.org
Vulkan 1.2.283 - Forward+ - Using Device #0: Apple - Apple M2 Pro

USER ERROR: .NET: Assemblies not found
   at: initialize (modules/mono/mono_gd/gd_mono.cpp:381)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.beta3.mono.official (82cedc83c9069125207c128f9a07ce3d82c317cc)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x00000001885de184 _sigtramp + 56
[2] Object* ClassDB::creator<MobileVRInterface>()
[3] Object* ClassDB::creator<MobileVRInterface>()
[4] JSON::get_data() const
[5] JSON::get_data() const
[6] JSON::get_data() const
[7] JSON::get_data() const
[8] JSON::get_data() const
[9] JSON::get_data() const
[10] JSON::get_data() const
[11] JSON::get_data() const
[12] JSON::get_data() const
[13] RendererCompositorRD::_create_current()
[14] RendererCompositorRD::_create_current()
[15] 15  dyld                                0x0000000188227274 start + 2840
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

1/ Create a C# project which includes an addon with C# assemblies
2/ Build the distribution for MacOS

### Minimal reproduction project (MRP)

[testblockchain.zip](https://github.com/user-attachments/files/16249124/testblockchain.zip)

**Comment by raulsntos** [comment]
- Sounds like it may be a duplicate of https://github.com/godotengine/godot/issues/94631

**Comment by raulsntos** [comment]
Since the linked issue has been fixed, can you check the [Godot 4.3 RC 1](https://godotengine.org/article/release-candidate-godot-4-3-rc-1/) release that includes the fix and see if that fixes it for you too?

**Comment by blockimperiumdao** [comment]
We are good. We can see our build assembling properly and launching without issue.
